File: solutions/169/169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int majorityElement = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            if (count == 0) {
                majorityElement = nums[i];
                count++;
            } else if (nums[i] == majorityElement) {
                count++;
            } else {
                count--;
            }
        }
        return majorityElement;
    }
};

File: solutions/169/169.py
class Solution(object):
    def majorityElement(self, nums):
        majorityElement = 0
        count = 0
        for i in range (len(nums)):
            if count == 0:
                majorityElement = nums[i]
            if nums[i] == majorityElement:
                count += 1
            else:
                count -= 1
        return majorityElement